### PR TITLE
TD-2968 fixing removing rejected sign off request

### PR DESCRIPTION
**Pull request by sherif-olaboye**
### JIRA link
https://hee-tis.atlassian.net/browse/TD-2968

### Description
need to add verify is null to only get only pending signoff request

### Screenshots
image

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors (see [info on Text Editor settings](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546185813/DLS+Dev+Process) to avoid whitespace changes)
- [ ] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript
- [ ] Tested any Views or partials created or changed with [Wave Chrome plugin](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh/related) and addressed any valid accessibility issues
- [ ] Updated/added documentation in [Confluence](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546939432/DLS+Code) and/or [GitHub Readme](https://github.com/TechnologyEnhancedLearning/DLSV2/blob/master/README.md). List of documentation links added/changed:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken
- [x] Scanned over my pull request in GitHub and addressed any warnings from the GitHub Build and Test checks.

### CODE REVIEW

File: DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/CandidateAssessmentsDataService.cs [hunk 1]
```diff
@@ -161,8 +161,10 @@ public void RemoveSignoffRequests(int selfAssessmentId, int delegateUserId, int
 						  ON CA.SelfAssessmentID = SAS.SelfAssessmentID
                     INNER JOIN CompetencyGroups AS CG 
 					ON SAS.CompetencyGroupID = CG.ID AND SAS.SelfAssessmentID =@selfAssessmentId
-                    WHERE ((ca.DelegateUserID = @delegateUserId) AND (ca.SelfAssessmentID = @selfAssessmentId) AND (sasr.SelfAssessmentReview = 1) AND (CG.ID =@competencyGroupId) AND (casv.SignedOff =0))
-                        OR ((ca.DelegateUserID = @delegateUserId) AND (ca.SelfAssessmentID = @selfAssessmentId) AND (sasr.SelfAssessmentReview IS NULL) AND (CG.ID =@competencyGroupId) AND (casv.SignedOff =0))
+                    WHERE ((ca.DelegateUserID = @delegateUserId) AND (ca.SelfAssessmentID = @selfAssessmentId) AND (sasr.SelfAssessmentReview = 1) AND
```
Comment (rshrirohit) [inline DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/CandidateAssessmentsDataService.cs:164]: as per the comment on jira, "We should be removing open requests (from CandidateAssessmentSupervisorVerifications) where Verified date is null, it looks like we are removing requests where SignedOff is 0 instead". We should remove the 
"(casv.SignedOff =0)" clause from here.

File: DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/CandidateAssessmentsDataService.cs [hunk 2]
```diff
@@ -161,8 +161,10 @@ public void RemoveSignoffRequests(int selfAssessmentId, int delegateUserId, int
 						  ON CA.SelfAssessmentID = SAS.SelfAssessmentID
                     INNER JOIN CompetencyGroups AS CG 
 					ON SAS.CompetencyGroupID = CG.ID AND SAS.SelfAssessmentID =@selfAssessmentId
-                    WHERE ((ca.DelegateUserID = @delegateUserId) AND (ca.SelfAssessmentID = @selfAssessmentId) AND (sasr.SelfAssessmentReview = 1) AND (CG.ID =@competencyGroupId) AND (casv.SignedOff =0))
-                        OR ((ca.DelegateUserID = @delegateUserId) AND (ca.SelfAssessmentID = @selfAssessmentId) AND (sasr.SelfAssessmentReview IS NULL) AND (CG.ID =@competencyGroupId) AND (casv.SignedOff =0))
+                    WHERE ((ca.DelegateUserID = @delegateUserId) AND (ca.SelfAssessmentID = @selfAssessmentId) AND (sasr.SelfAssessmentReview = 1) AND
+                           (CG.ID =@competencyGroupId) AND (casv.SignedOff =0) AND (casv.Verified IS NULL))
+                        OR ((ca.DelegateUserID = @delegateUserId) AND (ca.SelfAssessmentID = @selfAssessmentId) AND
+                        (sasr.SelfAssessmentReview IS NULL) AND (CG.ID =@competencyGroupId) AND (casv.SignedOff =0) AND (casv.Verified IS NULL))
```
Comment (rshrirohit) [inline DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/CandidateAssessmentsDataService.cs:167]: Same for here

**Review by rshrirohit — CHANGES_REQUESTED** [review]
Please make changes as requested.

File: DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/CandidateAssessmentsDataService.cs (hunk 2 above)
Comment (sherif-olaboye) [reply DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/CandidateAssessmentsDataService.cs:167]: I did not understand essence of this you comment

File: DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/CandidateAssessmentsDataService.cs (hunk 1 above)
Comment (sherif-olaboye) [reply DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/CandidateAssessmentsDataService.cs:164]: done

File: DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/CandidateAssessmentsDataService.cs (hunk 2 above)
Comment (sherif-olaboye) [reply DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/CandidateAssessmentsDataService.cs]: the clause have been removed

**Pull request merged**